src/main/java/com/igot/cb/interest/service/impl/InterestServiceImpl.java

```diff
@@ -203,7 +203,24 @@ public CustomResponse assignInterestToDemand(JsonNode interestDetails) {
           interestDetails.get(Constants.DEMAND_ID_RQST).asText());
       if (demandEntity.isPresent()) {
         JsonNode fetchedDemandJson = demandEntity.get().getData();
-        ((ObjectNode) fetchedDemandJson).put(Constants.STATUS, Constants.ASSIGNED);
+        if (fetchedDemandJson.get(Constants.STATUS).asText().equalsIgnoreCase(Constants.UNASSIGNED)){
```

Comment:
Need to add validation if json object is present then only do get

Reply:
Made the change